packages/expo-cli/src/commands/webhooks.ts

```diff
@@ -1,150 +1,171 @@
+async function listAsync(projectRoot: string) {
+  const { experienceName, project, client } = await setupAsync(projectRoot);
 
-  const event = _sanitizeEvent(_event);
-  if (!event) {
-    throw new Error('Webhook type has to be provided');
+  const webhooks = await client.getAsync(`projects/${project.id}/webhooks`);
+  if (webhooks.length) {
+    const table = new CliTable({ head: ['Webhook ID', 'URL', 'Event'] });
+    table.push(...webhooks.map((hook: Webhook) => [hook.id, hook.url, hook.event]));
+    log(table.toString());
+  } else {
+    log(`${chalk.bold(experienceName)} has no webhooks.`);
+    log('Use `expo webhooks:add` to add one.');
   }
+}
 
-  if (!url) {
-    throw new Error('You must provide --url parameter');
-  }
-  const isValidUrl = validator.isURL(url, {
-    protocols: ['http', 'https'],
-    require_protocol: true,
-  });
-  if (!isValidUrl) {
-    throw new Error(
-      'The provided webhook URL is invalid and must be an absolute URL, including a scheme.'
-    );
-  }
+async function addAsync(
+  projectRoot: string,
+  { url, event, ...options }: { url?: string; event?: WebhookEvent; secret?: string }
+) {
+  invariant(typeof url === 'string' && /^https?/.test(url), '--url: a HTTP URL is required');
+  invariant(typeof event === 'string', '--event: string is required');
+  const secret = validateSecret(options) || generateSecret();
 
-  if (secret) {
-    const secretString = String(secret);
-    if (secretString.length < 16 || secretString.length > 1000) {
-      throw new Error('Webhook secret has be at least 16 and not more than 1000 characters long');
-    }
-  } else {
-    secret = await _askForSecret();
-  }
+  const { experienceName, project, client } = await setupAsync(projectRoot);
 
-  return { url, secret, event };
+  const spinner = ora(`Adding webhook to ${experienceName}`).start();
+  await client.postAsync(`projects/${project.id}/webhooks`, { url, event, secret });
+  spinner.succeed();
 }
 
-function _sanitizeEvent(event: string | undefined, required = false): string | undefined {
-  if (!event) {
-    // we don't have anything to sanitize here, continue
-    return event;
-  }
+async function updateAsync(
+  projectRoot: string,
+  {
+    id,
+    url,
+    event,
+    ...options
+  }: { id?: string; url?: string; event?: WebhookEvent; secret?: string }
+) {
+  invariant(typeof id === 'string', '--id must be a webhook ID');
+  invariant(event == null || typeof event === 'string', '--event: string is required');
+  let secret = validateSecret(options);
+
+  const { project, client } = await setupAsync(projectRoot);
+
+  const webhook = await client.getAsync(`projects/${project.id}/webhooks/${id}`);
+  event = event ?? webhook.event;
+  secret = secret ?? webhook.secret;
+
+  const spinner = ora(`Updating webhook ${id}`).start();
+  await client.patchAsync(`projects/${project.id}/webhooks/${id}`, { url, event, secret });
+  spinner.succeed();
+}
```

Comment:
Seems useful to write a test here to make sure we're calling the right HTTP method (PATCH)